crates/cxx-qt-build/src/dir.rs

```diff
@@ -37,14 +42,26 @@ pub(crate) fn clean(path: impl AsRef<Path>) -> Result<()> {
 
 /// The target directory, namespaced by crate
 pub(crate) fn crate_target() -> PathBuf {
-    target().join("crates").join(crate_name())
+    let path = target();
+    if is_exporting_crate() {
+        path.join("crates").join(crate_name())
+    } else {
+        // If we're not exporting, use a shortened path
+        // The paths for files in the OUT_DIR can get pretty long, especially if combined with
+        // Corrosion/CMake.
+        // This is an issue, as Windows has a maximum path length of 260 characters.
```

Comment:
can we somewhere detect if we think the path you are building in is too long and warn on Windows?

Reply:
Lets do this later :-)